plugins/polio/tasks/api/refresh_lqas_data.py

```diff
@@ -0,0 +1,227 @@
+            )
+            return queryset.filter(query)
+
+        return queryset
+
+
+class RefreshLQASDataViewset(ModelViewSet):
+    permission_classes = [permissions.IsAuthenticated, HasPermission(permission.POLIO, permission.POLIO_CONFIG)]  # type: ignore
+    http_method_names = ["get", "post", "patch"]
+    model = Task
+    filter_backends = [
+        filters.OrderingFilter,
+        CustomTaskSearchFilterBackend,
+        DjangoFilterBackend,
+    ]
+    filterset_fields = {"created_at": ["gte"], "ended_at": ["exact"], "started_at": ["gte"], "status": ["exact"]}
+    ordering_fields = ["created_at", "ended_at", "name", "started_at", "status"]
+    ordering = ["-started_at"]
+
+    def get_serializer_class(self):
+        if self.request.method == "POST":
+            return RefreshLQASDataSerializer
+        return ExternalTaskSerializer
+
+    def get_queryset(self):
+        user = self.request.user
+        account = user.iaso_profile.account
+        queryset = Task.objects.filter(account=account).filter(external=True)
+        authorized_countries = user.iaso_profile.org_units.filter(org_unit_type_id__category="COUNTRY")
+        if authorized_countries.count() > 0:
+            authorized_names = [f"{TASK_NAME}-{id}" for id in authorized_countries]
+            queryset = queryset.filter(name__in=authorized_names)
+        return queryset
+
+    def create(self, request):
+        serializer = RefreshLQASDataSerializer(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        user = request.user
+        data = serializer.validated_data
+        started_at = datetime.now()
+        country_id = data.get("country_id", None)
+        name = f"{TASK_NAME}-{country_id}" if country_id is not None else TASK_NAME
+        task = Task.objects.create(
+            launcher=user,
+            account=user.iaso_profile.account,
+            name=name,
+            status=RUNNING,
+            external=True,
+            started_at=started_at,
+            should_be_killed=False,
+        )
+        status = self.refresh_lqas_data(country_id, task.id)
+        task.status = status
+        task.save()
+        return Response({"task": TaskSerializer(instance=task).data})
+
+    def refresh_lqas_data(self, country_id=None, task_id=None):
+        transport = RequestsHTTPTransport(
+            url=settings.OPENHEXA_URL,
+            verify=True,
+            headers={"Authorization": f"Bearer {settings.OPENHEXA_TOKEN}"},
+        )
+        client = Client(transport=transport, fetch_schema_from_transport=True)
+        get_runs = gql(
+            """
+        query pipeline {
+            pipeline(id: "%s"){
+                runs{
+                    items{
+                        run_id
+                        status
+                        config
+                    }
+                }
+            }
+        }
+        """
+            % (settings.LQAS_PIPELINE)
+        )
```

Comment:
Next time we add a connection to OpenHexa we should extract these calls into an OpenHexa API client "wrapper"

Reply:
Agreed